frontend/src/routes/Infrastructure/Clusters/ManagedClusters/components/DistributionField.tsx

```diff
@@ -1,29 +1,105 @@
 import { RbacButton } from '../../../../../components/Rbac'
-import { Cluster, ClusterStatus } from '../../../../../lib/get-cluster'
+import {
+    checkCuratorLatestFailedOperation,
+    checkCuratorLatestOperation,
+    Cluster,
+    ClusterStatus,
+    CuratorCondition,
+    getConditionStatusMessage,
+} from '../../../../../lib/get-cluster'
 import { rbacCreate, rbacPatch } from '../../../../../lib/rbac-util'
-import { ClusterCuratorDefinition } from '../../../../../resources/cluster-curator'
+import { AnsibleJob, getLatestAnsibleJob } from '../../../../../resources/ansible-job'
+import { ClusterCurator, ClusterCuratorDefinition } from '../../../../../resources/cluster-curator'
 import { BatchUpgradeModal } from './BatchUpgradeModal'
+
 export const backendUrl = `${process.env.REACT_APP_BACKEND_PATH}`
 
-export function DistributionField(props: { cluster?: Cluster }) {
+export function DistributionField(props: {
+    cluster?: Cluster
+    clusterCurator?: ClusterCurator | undefined
+    ansibleJobs?: AnsibleJob[] | undefined
+}) {
     const { t } = useTranslation(['cluster'])
     const [open, toggleOpen] = useState<boolean>(false)
     const toggle = () => toggleOpen(!open)
+    const clusterCurator = props.clusterCurator
+    const upgradeInfo = props.cluster?.distribution?.upgradeInfo
+    const ccConditions: V1CustomResourceDefinitionCondition[] = clusterCurator?.status?.conditions ?? []
+
+    let latestAnsibleJob: { prehook: AnsibleJob | undefined; posthook: AnsibleJob | undefined }
+    if (props.cluster?.namespace && props.ansibleJobs)
+        latestAnsibleJob = getLatestAnsibleJob(props.ansibleJobs, props.cluster?.namespace)
+    else latestAnsibleJob = { prehook: undefined, posthook: undefined }
 
     if (!props.cluster?.distribution) return <>-</>
     // use display version directly for non-online clusters
+
+    // Pre/Post hook
+    if (upgradeInfo?.isUpgradeCuration) {
+        // hook state
+        let statusType = StatusType.progress
+        let statusTitle = checkCuratorLatestOperation(CuratorCondition.posthook, ccConditions)
+            ? 'upgrade.ansible.posthookjob.title'
+            : 'upgrade.ansible.prehookjob.title'
+        let statusMessage = checkCuratorLatestOperation(CuratorCondition.posthook, ccConditions)
+            ? 'upgrade.ansible.posthook'
+            : 'upgrade.ansible.prehook'
+        let footerContent: ReactNode | string = (
+            <AcmButton
+                onClick={() => window.open(latestAnsibleJob.prehook?.status?.ansibleJobResult?.url)}
+                variant="link"
+                role="link"
+                icon={<ExternalLinkAltIcon />}
+                iconPosition="right"
+                isDisabled={!latestAnsibleJob.prehook?.status?.ansibleJobResult?.url}
+            >
+                {t('view.logs')}
+            </AcmButton>
+        )
+
+        // if pre/post failed
+        if (checkCuratorLatestFailedOperation(CuratorCondition.upgrade, ccConditions)) {
```

Comment:
We still have to do this, when the get-cluster file is attempting to extract the failures?

Reply:
I don't believe we have to, but checking the common case first (if there is a failure in either hook) cuts down on redundant code in the inner conditions (prehook/posthook); statusType and the footerContent will be declared the same way in either case, made sense to do it once then check each hook for the specific stuff.